evap/staff/views.py

```diff
@@ -404,6 +404,13 @@ def semester_delete(request):
 
     if not semester.can_be_deleted_by_manager:
         raise SuspiciousOperation("Deleting semester not allowed")
+    for course in semester.courses.all():
```

Comment:
Considering performance - can't we remove these two loops and make this O(1) database queries?
Like `RatingAnswerCounter.objects.filter(contribution__evaluation__course__semester=semester).delete()` and so on.